### PR TITLE
Add configurations of accessanalyzer (1), acmpca (2), appconfig (2), applicationinsights (1), apprunner (2), appsync (1), auditmanager (5), ce (3), cloudfront (1), cloudwatch (1), codepipeline (1), cognito (1), comprehend (2), connect (4), controltower (1) groups

### LINKED ISSUE #470: Configure accessanalyzer (1), acmpca (2), appconfig (2), applicationinsights (1), apprunner (2), appsync (1), auditmanager (5), ce (3), cloudfront (1), cloudwatch (1), codepipeline (1), cognito (1), comprehend (2), connect (4), controltower (1)
Configure external name without testing:

* `accessanalyzer`
  * [x] [aws_accessanalyzer_archive_rule](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/accessanalyzer_archive_rule)

* `acmpca`
  * [x] [aws_acmpca_permission](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acmpca_permission)
  * [x] [aws_acmpca_policy](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acmpca_policy)

* `appconfig`
  * [x] [aws_appconfig_extension](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_extension)
  * [x] [aws_appconfig_extension_association](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_extension_association)

* `applicationinsights`
  * [x] [aws_applicationinsights_application](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/applicationinsights_application)

* `apprunner`
  * [x] [aws_apprunner_observability_configuration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/apprunner_observability_configuration)
  * [x] [aws_apprunner_vpc_ingress_connection](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/apprunner_vpc_ingress_connection)

* `appsync`
  * [x] [aws_appsync_type](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appsync_type)

* `auditmanager`
  * [x] [aws_auditmanager_account_registration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/auditmanager_account_registration)
  * [x] [aws_auditmanager_assessment](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/auditmanager_assessment)
  * [x] [aws_auditmanager_assessment_report](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/auditmanager_assessment_report)
  * [x] [aws_auditmanager_control](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/auditmanager_control)
  * [x] [aws_auditmanager_framework](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/auditmanager_framework)

* `ce`
  * [x] [aws_ce_anomaly_monitor](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ce_anomaly_monitor)
  * [x] [aws_ce_anomaly_subscription](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ce_anomaly_subscription)
  * [x] [aws_ce_cost_allocation_tag](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ce_cost_allocation_tag)

* `cloudfront`
  * [x] [aws_cloudfront_origin_access_control](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_origin_access_control)

* `cloudwatch`
  * [x] [aws_cloudwatch_log_data_protection_policy](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_log_data_protection_policy)

* `codepipeline`
  * [x] [aws_codepipeline_custom_action_type](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/codepipeline_custom_action_type)

* `cognito`
  * [x] [aws_cognito_risk_configuration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cognito_risk_configuration)

* `comprehend`
  * [x] [aws_comprehend_document_classifier](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/comprehend_document_classifier)
  * [x] [aws_comprehend_entity_recognizer](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/comprehend_entity_recognizer)

* `connect`
  * [x] [aws_connect_instance_storage_config](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_instance_storage_config)
  * [x] [aws_connect_phone_number](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_phone_number)
  * [x] [aws_connect_user](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_user)
  * [x] [aws_connect_vocabulary](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_vocabulary)

* `controltower`
  * [x] [aws_controltower_control](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/controltower_control)

### PR DESCRIPTION
**Pull request by dverveiko**
### Description of your changes
Add external name configuration without testing:

* `accessanalyzer`
  * [x] [aws_accessanalyzer_archive_rule](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/accessanalyzer_archive_rule)

* `acmpca`
  * [x] [aws_acmpca_permission](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acmpca_permission)
  * [x] [aws_acmpca_policy](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acmpca_policy)

* `appconfig`
  * [x] [aws_appconfig_extension](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_extension)
  * [x] [aws_appconfig_extension_association](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appconfig_extension_association)

* `applicationinsights`
  * [x] [aws_applicationinsights_application](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/applicationinsights_application)

* `apprunner`
  * [x] [aws_apprunner_observability_configuration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/apprunner_observability_configuration)
  * [x] [aws_apprunner_vpc_ingress_connection](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/apprunner_vpc_ingress_connection)

* `appsync`
  * [x] [aws_appsync_type](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appsync_type)

* `auditmanager`
  * [x] [aws_auditmanager_account_registration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/auditmanager_account_registration)
  * [x] [aws_auditmanager_assessment](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/auditmanager_assessment)
  * [x] [aws_auditmanager_assessment_report](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/auditmanager_assessment_report)
  * [x] [aws_auditmanager_control](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/auditmanager_control)
  * [x] [aws_auditmanager_framework](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/auditmanager_framework)

* `ce`
  * [x] [aws_ce_anomaly_monitor](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ce_anomaly_monitor)
  * [x] [aws_ce_anomaly_subscription](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ce_anomaly_subscription)
  * [x] [aws_ce_cost_allocation_tag](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ce_cost_allocation_tag)

* `cloudfront`
  * [x] [aws_cloudfront_origin_access_control](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_origin_access_control)

* `cloudwatch`
  * [x] [aws_cloudwatch_log_data_protection_policy](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_log_data_protection_policy)

* `codepipeline`
  * [x] [aws_codepipeline_custom_action_type](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/codepipeline_custom_action_type)

* `cognito`
  * [x] [aws_cognito_risk_configuration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cognito_risk_configuration)

* `comprehend`
  * [x] [aws_comprehend_document_classifier](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/comprehend_document_classifier)
  * [x] [aws_comprehend_entity_recognizer](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/comprehend_entity_recognizer)

* `connect`
  * [x] [aws_connect_instance_storage_config](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_instance_storage_config)
  * [x] [aws_connect_phone_number](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_phone_number)
  * [x] [aws_connect_user](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_user)
  * [x] [aws_connect_vocabulary](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_vocabulary)

* `controltower`
  * [x] [aws_controltower_control](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/controltower_control)

Fixes #470 

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### CODE REVIEW

File: config/externalnamenottested.go [hunk 1]
```diff
@@ -590,4 +590,115 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+	"aws_applicationinsights_application": config.ParameterAsIdentifier("resource_group_name"),
+
+	// apprunner
+	//
+	// App Runner Observability Configuration can be imported by using the arn
+	// Example: arn:aws:apprunner:us-east-1:1234567890:observabilityconfiguration/example/1/d75bc7ea55b71e724fe5c23452fe22a1
+	"aws_apprunner_observability_configuration": config.IdentifierFromProvider,
+	// App Runner VPC Ingress Connection can be imported by using the arn
+	// Example: arn:aws:apprunner:us-west-2:837424938642:vpcingressconnection/example/b379f86381d74825832c2e82080342fa
+	"aws_apprunner_vpc_ingress_connection": config.IdentifierFromProvider,
```
Comment (ulucinar) [inline config/externalnamenottested.go:627]: Similar to the discussion above, but this time we do not have an annoying unknown component after the connection name. So, `name` looks like to be a naming field and we have a generated part (the last component). So, we could use (from the above example ARN) `example/b379f86381d74825832c2e82080342fa` as the external-name and treat `name` (Terraform configuration) argument as a naming field (i.e., remove it from `spec.forProvider`). 

One similar example is here:
https://github.com/upbound/provider-aws/blob/4194064bf89b91bae2f4739fab7d1245485b0980/config/externalname.go#L1527
for the `aws_cloudformation_stack` resource. But in this case, looks like we just normalized the external-name but still kept the naming argument `spec.forProvider.name`.

TLDR; we may at least normalize the external-name here by following a similar approach with `aws_cloudformation_stack`'s configuration. But we had better also have a TODO so that with some further effort, we can also normalize the `spec.forProvider` API.

File: config/externalnamenottested.go [hunk 2]
```diff
@@ -590,4 +590,115 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+
+	// appsync
+	//
+	// Appsync Types can be imported using the id (api-id:format:name)
+	"aws_appsync_type": config.IdentifierFromProvider,
```
Comment (ulucinar) [inline config/externalnamenottested.go:632]: Not sure what `name` that appears in the Terraform ID is (maybe the API name (of the referenced API), not sure). We may at least be able to normalize the external-name with something like:
```suggestion
	"aws_appsync_type": config.TemplatedStringAsIdentifier("", "{{ .parameters.api_id }}:{{ .parameters.format }}:{{ .external_name }}"),
```

This would just leave `name` as the value of the external-name annotation. Not sure if we have a naming field (thus an opportunity for `spec.forProvider` normalization here).

File: config/externalnamenottested.go [hunk 3]
```diff
@@ -590,4 +590,115 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+	// Audit Manager Account Registration resources can be imported using the id
+	"aws_auditmanager_account_registration": config.IdentifierFromProvider,
+	// Audit Manager Assessments can be imported using the assessment id (abc123-de45)
+	"aws_auditmanager_assessment": config.IdentifierFromProvider,
+	// Audit Manager Assessment Reports can be imported using the assessment report id (abc123-de45)
+	"aws_auditmanager_assessment_report": config.IdentifierFromProvider,
+	// An Audit Manager Control can be imported using the id (abc123-de45)
+	"aws_auditmanager_control": config.IdentifierFromProvider,
+	// Audit Manager Framework can be imported using the framework id (abc123-de45)
+	"aws_auditmanager_framework": config.IdentifierFromProvider,
```
Comment (ulucinar) [inline config/externalnamenottested.go:645]: Let's make sure the `name` argument does not really appear in the ID for these resources (during the tests). If it appears, then we can treat it as a naming field. But with the current information we have, `config.IdentifierFromProvider` is the only option we have (and leave `spec.forProvider.name`).

File: config/externalnamenottested.go [hunk 4]
```diff
@@ -590,4 +590,115 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+
+	// ce
+	//
+	// aws_ce_anomaly_monitor can be imported using the id
+	"aws_ce_anomaly_monitor": config.IdentifierFromProvider,
+	// aws_ce_anomaly_subscription can be imported using the id
+	"aws_ce_anomaly_subscription": config.IdentifierFromProvider,
```
Comment (ulucinar) [inline config/externalnamenottested.go:652]: Let's add TODOs (for tests) here. We may have a case of normalization when we observe the ARN syntax for these resources.

File: config/externalnamenottested.go [hunk 5]
```diff
@@ -590,4 +590,115 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+	// aws_ce_cost_allocation_tag can be imported using the id
+	"aws_ce_cost_allocation_tag": config.IdentifierFromProvider,
```
Comment (ulucinar) [inline config/externalnamenottested.go:654]: I would suggest treating `tag_key` as the naming argument and the ID for this resource here and validating this during the tests:
```suggestion
	"aws_ce_cost_allocation_tag": config.ParameterAsIdentifier("tag_key"),
```

File: config/externalnamenottested.go [hunk 6]
```diff
@@ -590,4 +590,115 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+	"aws_comprehend_document_classifier": config.TemplatedStringAsIdentifier("name", "arn:aws:comprehend:{{ .setup.configuration.region }}:{{ .setup.client_metadata.account_id }}:document_classifier/{{ .external_name }}"),
+	// Comprehend Entity Recognizer can be imported using the ARN
+	// Example: arn:aws:comprehend:us-west-2:123456789012:entity-recognizer/example
+	"aws_comprehend_entity_recognizer": config.TemplatedStringAsIdentifier("name", "arn:aws:comprehend:{{ .setup.configuration.region }}:{{ .setup.client_metadata.account_id }}:entity-recognizer/{{ .external_name }}"),
+
+	// connect
+	//
+	// Amazon Connect Instance Storage Configs can be imported using the instance_id, association_id, and resource_type separated by a colon (:)
+	// Example: f1288a1f-6193-445a-b47e-af739b2:c1d4e5f6-1b3c-1b3c-1b3c-c1d4e5f6c1d4e5:CHAT_TRANSCRIPTS
+	"aws_connect_instance_storage_config": config.IdentifierFromProvider,
```
Comment (ulucinar) [inline config/externalnamenottested.go:689]: Looks like we can only leave the association id in the external-name (external-name normalization). The following may work (to be verified with the tests):
```suggestion
	"aws_connect_instance_storage_config": config.TemplatedStringAsIdentifier("", "{{ .parameters.instance_id }}:{{ .external_name }}:{{ .parameters.resource_type }}"),
```

File: config/externalnamenottested.go [hunk 7]
```diff
@@ -590,4 +590,115 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+	// Amazon Connect Phone Numbers can be imported using its id
+	"aws_connect_phone_number": config.IdentifierFromProvider,
+	// Amazon Connect Users can be imported using the instance_id and user_id separated by a colon (:)
+	// Example: f1288a1f-6193-445a-b47e-af739b2:c1d4e5f6-1b3c-1b3c-1b3c-c1d4e5f6c1d4e5
+	"aws_connect_user": config.IdentifierFromProvider,
```
Comment (ulucinar) [inline config/externalnamenottested.go:694]: We can try just leaving the `user_id` in the external-name:
```suggestion
	"aws_connect_user": config.TemplatedStringAsIdentifier("", "{{ .parameters.instance_id }}:{{ .external_name }}"),
```

File: config/externalnamenottested.go [hunk 8]
```diff
@@ -590,4 +590,115 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+	// Amazon Connect Vocabularies can be imported using the instance_id and vocabulary_id separated by a colon (:)
+	// Example: f1288a1f-6193-445a-b47e-af739b2:c1d4e5f6-1b3c-1b3c-1b3c-c1d4e5f6c1d4e5
+	"aws_connect_vocabulary": config.IdentifierFromProvider,
```
Comment (ulucinar) [inline config/externalnamenottested.go:697]: We can try just leaving the `vocabulary_id` in the external-name:
```suggestion
	"aws_connect_user": config.TemplatedStringAsIdentifier("", "{{ .parameters.instance_id }}:{{ .external_name }}"),
```

File: config/externalnamenottested.go [hunk 9]
```diff
@@ -590,4 +590,115 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
...
+
+	// controltower
+	//
+	// Control Tower Controls can be imported using their organizational_unit_arn/control_identifier
+	// Example: arn:aws:organizations::123456789101:ou/o-qqaejywet/ou-qg5o-ufbhdtv3,arn:aws:controltower:us-east-1::control/WTDSMKDKDNLE
+	"aws_controltower_control": config.TemplatedStringAsIdentifier("", "{{ .parameters.target_identifier }},{{ .parameters.control_identifier }}"),
```
Comment (ulucinar) [inline config/externalnamenottested.go:703]: Although this configuration looks correct to me, the end result would be the same as using `config.IdentifierFromProvider` (no external-name or `spec.forProvider` normalization). What we could try instead is:
```suggestion
	"aws_controltower_control": config.TemplatedStringAsIdentifier("", "{{ .parameters.target_identifier }},{{ .external_name }}"),
```
, which should leave only the `control_identifier` in the external-name (if this works) and thus we would at least normalize the external-name.

**Review by ulucinar — APPROVED** [review]
Thank you @dverveiko, lgtm.
